core/sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/constraint/Operation.java

```diff
@@ -23,7 +23,7 @@
 	Operation(SparqlOperator operator, int operandLimit) {
 		super(operator);
 		this.operandLimit = operandLimit;
-		if (operator == ConnectiveOperator.AND) {
+		if (operator == ConnectiveOperator.AND || operator == ConnectiveOperator.OR) {
```

Comment:
We noticed that the Paranthesize function needs to be called for the OR operator as well. This is because for queries which have an OR inside an AND, the query output is not how it should be. It does not take into consideration that the OR part must be evaluated first. 
However, calling the Paranthesize function as we have done here is giving us extra parenthesis inside the filter query whenever we have boolean binary operators (AND/OR). Is that fine? If not, kindly help us with how we should remove it. Although considering it evaluation wise, it will always give the correct output. Just that the brackets there are not essential.

Reply:
I'm ok with the extra parenthesis. I think it would be hard to fix and it's not semantically incorrect, just superfluous.

Reply:
I am wondering if all connective operators need parenthesises around them? Do you know @abhishekag03 ?

Reply:
Yeah, you're actually right! We need to include it for all Connective Operators. We will add the code for the same along with test cases for each operator. I'm just wondering if this should be done in this issue itself or shall we open a new issue and commit the changes there? Or can the issue heading be changed? Please advise what shall be done?

Reply:
It’ll be easier to do it in this PR and just update the title and description of both the issue and this PR.

Reply:
Sure, we have done the same and pushed the code. Since the check had to be done for all connective operators, we have now used `instanceOf` instead of equality check. Kindly have a look and let us know if something needs to be changed.